w5_doubly_linked_list/isaac/CircularyDoublyLinkedList.java

```diff
@@ -0,0 +1,137 @@
+package dataStructure.linkedList.doubly;
+
+public class CircularyDoublyLinkedList {
+    private Node head;
+    private Integer size;
+    public CircularyDoublyLinkedList(){
+        this.head = new Node();
+        this.size = 0;
+    }
+
+    public int getSize(){
+        return this.size;
+    }
+
+    public Node getHead(){
+        return this.head;
+    }
+
+    public Node getFirstNode(){
+        return this.head.getNext();
+    }
+
+    public void splice(Node a, Node b, Node x){
+        Node ap, bn, xn;
+
+        if(a.getPrev() == null && a.getNext() == null) {
+            ap = x;
+            bn = x.getNext();
+
+            ap.setNext(a);
+            a.setPrev(ap);
+            bn.setPrev(a);
+            a.setNext(bn);
+        } else {
+            ap = a.getPrev();
+            bn = b.getNext();
+            xn = x.getNext();
+
+            ap.setNext(bn);
+            bn.setPrev(ap);
+            x.setNext(a);
+            a.setPrev(x);
+            xn.setPrev(b);
+            b.setNext(xn);
+        }
+    }
+
+    // a를 x노드 뒤로 이동한다.
+    public void moveAfter(Node a, Node x){
+        splice(a, a, x);
+    }
+
+    public void moveBefore(Node a, Node x){
+        splice(a, a, x.getPrev());
+    }
+
+    public void insertAfter(Node x, String key){
+        moveAfter(new Node(key), x);
+        this.size++;
+    }
+
+    public void insertBefore(Node x, String key){
+        moveBefore(new Node(key), x);
+        this.size++;
+    }
+
+    public void pushFront(String key){
+        insertBefore(this.head.getNext(), key);
```

Comment:
저는 `insertBefore`를 사용하면 `this.head.getNext()` 요렇게 `getNext()`까지 한 번 더 생각해야 해서 조금 더 햇가리는 것 같아요 🤔 

아래처럼 `insertAfter`로 처리해보는건 어떠신가요?
```suggestion
        insertAfter(this.head, key);
```